src/trace-plugin-interface.js

```diff
@@ -0,0 +1,244 @@
+ * collecting call stack information for the root span, starting from the top;
+ * this should be set to avoid including frames in the PluginAPI. Defaults to 0.
+ * @returns A new Transaction object, or null if the trace agent's policy has
+ * disabled tracing for the given set of options.
+ */
+PluginAPI.prototype.createTransaction = function(options) {
+  options = options || {};
+  var that = this;
+  // If the options object passed in has the getHeader field set,
+  // try to retrieve the header field containing incoming trace metadata.
+  var incomingTraceContext;
+  if (is.fn(options.getHeader)) {
+    var header = options.getHeader('x-cloud-trace-context');
+    if (header) {
+      incomingTraceContext = that.agent.parseContextFromHeader(header);
+    }
+  }
+  incomingTraceContext = incomingTraceContext || {};
+  if (options.url && !that.agent.shouldTrace(options.url, incomingTraceContext.options)) {
+    return null;
+  }
+  var rootContext = that.agent.createRootSpanData(options.name,
+    incomingTraceContext.traceId,
+    incomingTraceContext.spanId,
+    options.skipFrames ? options.skipFrames + 1 : 1);
+  // If the options object passed in has the setHeader field set,
+  // use it to set trace metadata in an outgoing request.
+  if (is.fn(options.setHeader)) {
+    var outgoingTraceContext = rootContext.traceId + '/' +
+      rootContext.spanId;
+    var outgoingHeaderOptions = (incomingTraceContext.options !== null &&
+      incomingTraceContext.options !== undefined) ?
+      incomingTraceContext.options : constants.TRACE_OPTIONS_TRACE_ENABLED;
+    outgoingTraceContext += (';o=' + outgoingHeaderOptions);
+    options.setHeader('x-cloud-trace-context', outgoingTraceContext);
+  }
+  return new Transaction(that.agent, rootContext);
+};
+
+PluginAPI.prototype.getTransaction = function() {
+  if (cls.getRootContext()) {
+    return new Transaction(this.agent, cls.getRootContext());
+  } else {
+    return null;
+  }
+};
+
+/**
+ * Runs the given function in a root span corresponding to an incoming request,
+ * possibly passing it an object that exposes an interface for adding labels
+ * and closing the span.
+ * @param {object} options An object that specifies options for how the root
+ * span is created and propogated. @see PluginAPI.prototype.createTransaction
+ * @param {function(?Transaction)} fn A function that will be called exactly
+ * once. If the incoming request should be traced, a root span will be created,
+ * and this function will be called with a Transaction object exposing functions
+ * operating on the root span; otherwise, it will be called without any
+ * arguments.
+ * @returns The return value of calling fn.
+ */
+PluginAPI.prototype.runInRootSpan = function(options, fn) {
+  var that = this;
+  return this.agent.namespace.runAndReturn(function() {
+    var oldSkipFrames = options.skipFrames;
+    options.skipFrames = options.skipFrames ? options.skipFrames + 2 : 2;
```

Comment:
I've changed it so that this (and the equivalent function for child spans) call on a private function that accepts `skipFrames` separately. This allows us not to create/mutate an `options` object at all, in exchange for some clutter in the interface file (though this can be folded away in a separate file).